File: app/src/main/java/org/simon/npo/controller/UseNpoController.java
package org.simon.npo.controller;

import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.simon.npo.dto.StartNpoRequest;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/v1/user-npo")
@RequiredArgsConstructor
public class UseNpoController {

  @PostMapping("/{warehouseId}/start/list")
  public ResponseEntity<Void> start(
      @PathVariable String warehouseId, @Valid StartNpoRequest request) {
    throw new UnsupportedOperationException();
  }
}

File: app/src/main/kotlin/org/simon/npo/controller/UseNpoController.kt
package org.simon.npo.controller

import jakarta.validation.Valid
import jakarta.validation.constraints.NotBlank
import lombok.RequiredArgsConstructor
import org.simon.npo.dto.UserNpoResponse
import org.simon.npo.dto.UserNpoStartRequest
import org.simon.npo.service.UserNpoService
import org.springframework.http.ResponseEntity
import org.springframework.validation.annotation.Validated
import org.springframework.web.bind.annotation.*


@RestController
@RequestMapping("/v1/user-npo")
@RequiredArgsConstructor
@Validated
class UseNpoController(private val userNpoService: UserNpoService) {

    @PostMapping("/{warehouseId}/start")
    fun start(
        @PathVariable warehouseId: String, @Valid @RequestBody request: UserNpoStartRequest
    ): ResponseEntity<Void> = userNpoService.start(warehouseId, request).run { ResponseEntity.ok().build() }


    @GetMapping("/{warehouseId}/active/{userName}")
    fun getActive(
        @PathVariable warehouseId: String, @PathVariable userName: @NotBlank String
    ): ResponseEntity<UserNpoResponse> = userNpoService.getActive(warehouseId, userName)
        ?.let { ResponseEntity.ok(it) }
        ?: ResponseEntity.noContent().build()
}
